ppb/sprites.py

```diff
@@ -141,228 +133,127 @@ def rotate(self, degrees):
         self.rotation += degrees
 
 
-class Side(FauxFloat):
-    """
-    Acts like a float, but also has a variety of accessors.
+class RectangleShapeMixin:
     """
-    sides = {
-        LEFT: ('x', -1),
-        RIGHT: ('x', 1),
-        TOP: ('y', 1),
-        BOTTOM: ('y', -1)
-    }
+    A Mixin that provides a rectangular area to sprites.
 
-    def __init__(self, parent: 'SquareShapeMixin', side: str):
-        self.side = side
-        self.parent = parent
+    You should include RectangleShapeMixin before your BaseSprite in your
+    parent classes.
+
+    Classes derived from RectangleShapeMixin default to the same size and
+    shape as all ppb Sprites: A 1 game unit by 1 game unit square. Just set
+    the width and height in your constructor (Or as class attributes) to
+    change this default.
+    """
+    width: int = 1
+    height: int = 1
+    # Following class properties for type hinting only. Your concrete sprite
+    # should already have one.
+    position: Vector
 
-    def __repr__(self):
-        return f"Side({self.parent!r}, {self.side!r})"
+    @property
+    def left(self) -> float:
+        return self.position.x - self.width / 2
 
-    def __str__(self):
-        return str(float(self))
+    @left.setter
+    def left(self, value: Union[float, int]):
+        self.position = Vector(value + (self.width * 0.5), self.position.y)
```

Comment:
Why `* 0.5` instead of `/ 2`?

Reply:
No particular reason, willing to change.